process-mining/src/com/axonivy/processmining/service/IvyTaskOccurrenceService.java

```diff
@@ -0,0 +1,106 @@
+		HashMap<String, TaskOccurrence> taskOccurrenceMap = getHashMapTaskOccurrencesByProcessId(processId);
+		HashMap<String, Integer> result = correctTaskOccurrences(taskOccurrenceMap);
+		// TODO: This block code is used for verification
+		// We will remove it when verification is finished
+		for (Map.Entry<String, Integer> taskOccurrence : result.entrySet()) {
+			Ivy.log().warn("Task {0} occurs {1} times", taskOccurrence.getKey(), taskOccurrence.getValue());
+		}
+
+		return result;
+	}
+
+	private static HashMap<String, TaskOccurrence> getHashMapTaskOccurrencesByProcessId(String processId) {
+		return Sudo.get(() -> {
+			TaskQuery taskQuery = TaskQuery.create().where().requestPath()
+					.isLike(String.format(LIKE_TEXT_SEARCH, processId)).orderBy().startTaskSwitchEventId();
+			HashMap<String, TaskOccurrence> map = new HashMap<>();
+			countTaskOccurrencesByTaskQuery(map, taskQuery);
+			return map;
+		});
+	}
+
+	private static void countTaskOccurrencesByTaskQuery(HashMap<String, TaskOccurrence> map, TaskQuery taskQuery) {
+		List<ITask> tasks = new ArrayList<>();
+		int maxQueryResults = Integer.valueOf(Ivy.var().get(IvyVariable.MAX_QUERY_RESULTS.getVariableName()));
+		Integer startIndex = 0;
+		do {
+			tasks = Ivy.wf().getTaskQueryExecutor().getResults(taskQuery, startIndex, maxQueryResults);
+			countTaskOccurrences(map, tasks);
+			startIndex += maxQueryResults;
+		} while (maxQueryResults == tasks.size());
+	}
+
+	private static void countTaskOccurrences(HashMap<String, TaskOccurrence> taskOccurrenceMap, List<ITask> tasks) {
+		for (ITask iTask : tasks) {
+			String taskElementId = getTaskElementIdFromRequestPath(iTask.getRequestPath());
+			updateTaskOccurrencesMap(taskOccurrenceMap, taskElementId, iTask.getStartSwitchEvent().getId());
+		}
+	}
+
+	private static String getTaskElementIdFromRequestPath(String requestPath) {
+		String[] arr = requestPath.split(SLASH);
+		// Request Path contains: {PROCESS ID}/.../{NAME OF TASK}
+		// So we have get the node before /{NAME OF TASK}
+		// Ignore case {PROCESS ID}/{NAME OF TASK}
+		return arr.length > 2 ? arr[arr.length - 2] : StringUtils.EMPTY;
+	}
+
+	private static void updateTaskOccurrencesMap(HashMap<String, TaskOccurrence> taskOccurrenceMap,
+			String taskElementId, Long startTaskSwitchEventId) {
+		if (StringUtils.isNotBlank(taskElementId)) {
+			TaskOccurrence taskOccurrence = getCountedTaskOccurrence(taskOccurrenceMap, taskElementId,
+					startTaskSwitchEventId);
+			taskOccurrenceMap.put(taskElementId, taskOccurrence);
+		}
+	}
+
+	private static TaskOccurrence getCountedTaskOccurrence(HashMap<String, TaskOccurrence> taskOccurrenceMap,
+			String taskElementId, Long startTaskSwitchEventId) {
+		TaskOccurrence taskOccurrence = taskOccurrenceMap.get(taskElementId);
+		if (taskOccurrence != null) {
+			if (startTaskSwitchEventId != null
+					&& taskOccurrence.getStartSwitchEventId().longValue() != startTaskSwitchEventId.longValue()) {
```

Comment:
Using equals to compare Long instead of converting it to 'long' with == operation